src/twisted/web/http_headers.py

```diff
@@ -81,6 +73,10 @@ class Headers:
         b"x-xss-protection": b"X-XSS-Protection",
     }
 
+    _canonicalHeaderCache: Dict[Union[bytes, str], bytes] = {}
+
+    __slots__ = ["_rawHeaders"]
```

Comment:
If we can compatibly go from non-slotted to slotted (can we?) could we just go full-blown `@dataclass` and save ourselves some headaches here, getting rid of the custom `@comparable`?

Reply:
The `@dataclass` made it slower (assuming you mean the attrs one).

The backwards incompatibility is you can't add random attributes to the class, which ... you really shouldn't. Subclasses can still add whatever attributes they want. So it seems fine to me.

Reply:
Actually `@dataclass` is fine, never mind.

Reply:
But also it has issues with `__init__`, and its the deprecated attrs API, and I don't really want to spend the time figuring this out.